crates/evm/fuzz/src/strategies/state.rs

```diff
@@ -245,12 +245,26 @@ impl FuzzDictionary {
             // Insert push bytes
             if let Some(code) = &account_info.code {
                 self.insert_address(*address);
-                self.collect_push_bytes(code.bytes_slice());
+                self.collect_push_bytes(code);
             }
         }
     }
 
-    fn collect_push_bytes(&mut self, code: &[u8]) {
+    fn collect_push_bytes(&mut self, code: &Bytecode) {
+        let mut code = code.original_byte_slice();
+        if code.is_empty() {
+            return;
+        }
+
+        // Remove metadata by looking up the last two bytes of original bytecode.
+        if code.len() > 2 {
+            let metadata_len = &code[code.len() - 2..];
+            let metadata_len = u16::from_be_bytes([metadata_len[0], metadata_len[1]]).into();
+            if code.len() > metadata_len {
```

Comment:
this could still fail due to the -2, but not sure if possible

perhaps `metadata_offset = metadata_len + 2`

&code[..code.len() - metadata_offset];